### DISCUSSION TITLE
A question for radationIRF.m script

### DISCUSSION THREAD

**Discussion by Nirvashcast** [WEC-Sim/WEC-Sim#1344] [Q&A] [answered]
Hello，this is very nice simulation toolbox and I  have a question about the code in radationIRF.m. The equation of coefficience ra_K is described as the following picture, which equals the integral of the product of coefficients B(ω) and cos(ωt) with respect to ω ，however，I notice that in the script, the Integral term is B(ω)·cos(ωt) ·ω. So I wonder if it is a mistake or if I have misunderstanded. I am very appreciate it if I can receive reply!
![K](https://github.com/user-attachments/assets/e6744ed1-ae65-4249-984e-18aa705d7acb)
![KinScript](https://github.com/user-attachments/assets/0882e881-f474-46ad-b8ba-720fe5cdd07c)

**Comment by MShabara** [comment, marked as answer]
Thank you for your question.

The function radiationIRF.m calculates the normalized radiation impulse response function, K̅(t), not the radiation impulse response function, K(t).

image

In the line `(2/pi)*trapz(w, ra_B .* (cos(w .* t(:)) .* w), 2)`, the variable `ra_B` represents the normalized radiation damping coefficient. The normalization is done by dividing the radiation damping coefficient by the fluid density (ρ) and the angular frequency (ω). Therefore, the equation includes a multiplication by ω to properly account for this normalization.

This is also mentioned in the first comment within the function. The normalization of the retardation function is achieved by dividing it by the fluid density (ρ).

**Comment by Nirvashcast** [comment]
Thank you very much for your reply and guide! I will read more carefully.




Nirvash
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "WEC-Sim/WEC-Sim"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年10月22日(星期二) 凌晨1:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [WEC-Sim/WEC-Sim] A question for radationIRF.m script (Discussion #1344)





 
Thank you for your question.
 
The function radiationIRF.m calculates the normalized radiation impulse response function, K̅(t), not the radiation impulse response function, K(t).
 
image
 
In the line (2/pi)*trapz(w, ra_B .* (cos(w .* t(:)) .* w), 2), the variable ra_B represents the normalized radiation damping coefficient. The normalization is done by dividing the radiation damping coefficient by the fluid density (ρ) and the angular frequency (ω). Therefore, the equation includes a multiplication by ω to properly account for this normalization.
 
This is also mentioned in the first comment within the function. The normalization of the retardation function is achieved by dividing it by the fluid density (ρ).

**Reply by MShabara** [reply]
It happens to the best of us, and these units can be confusing sometimes.
Free to post any questions you have, we will be happy to help.
